### ISSUE TITLE
Field Caps failing if using an alias pointing to a closed index

### ISSUE THREAD

**Issue by timroes** [elastic/elasticsearch#77612]
Currently when using an index pattern to get field caps over a set of indices where some might be closes works completely fine. If you use an alias for those indexes instead the _field_caps call suddenly fails.

Requests to reproduce this:

```
PUT /closed-test-1
{
  "settings": {
    "number_of_shards": 1
  },
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      }
    }
  },
  "aliases": {
    "alert": {}
  }
}

PUT /closed-test-2
{
  "settings": {
    "number_of_shards": 1
  },
  "mappings": {
    "properties": {
      "@timestamp": {
        "type": "date"
      }
    }
  },
  "aliases": {
    "alert": {}
  }
}

POST closed-test-1/_close

GET closed-test-*/_field_caps?fields=*
GET alert/_field_caps?fields=*
```

### Actual behavior

The field_caps call against `closed-test-*` works perfecty fine and returns a result for only `closed-test-2`. The _field_caps call against `alert` will fail with:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "cluster_block_exception",
        "reason" : "index [closed-test-1] blocked by: [FORBIDDEN/4/index closed];"
      }
    ],
    "type" : "cluster_block_exception",
    "reason" : "index [closed-test-1] blocked by: [FORBIDDEN/4/index closed];"
  },
  "status" : 403
}

```

### Expected behavior

Both field caps calls should behave the same and return the same result in this example. The 2nd field_caps call should not fail.

Related Kibana issue: https://github.com/elastic/kibana/issues/111699

**Comment by jimczi** [comment]
This is a known issue that affects lots of APIs, closing as duplicate of #32238.
